test/runFIT.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash -ex
+
+############################################
+# Global variable
+############################################
+RACKHD_TEST_DIR="$( cd -P "$( dirname "$0" )" && pwd )"
+
+#############################################
+#
+# Usage
+############################################
+Usage(){
+    set +x
+    echo "Function: This script is used to set up environment for FIT and run FIT."
+    echo "Usage: $0 [OPTIONS]"
+    echo "  OPTIONS:"
+    echo "    Mandatory options:"
+    echo "      -w, --WORKSPACE: The directory of workspace( where the code will be cloned to and staging folder), it's required"
+    echo "      -p, --PASSWORD: password of current user which is used for FIT to log into host by, it's required."
+    echo "    Optional options:"
+    echo "      -g, --TEST_GROUP: test group of FIT, such as imageservice, smoke"
+    set -x
+}
+
+
+#############################################
+#
+#  Create the virtual env for FIT  
+#
+############################################
+setupVirtualEnv(){
+    pushd ${RACKHD_TEST_DIR}
+    virtual_env_name=FIT
+    rm -rf .venv/$virtual_env_name
+    ./mkenv.sh $virtual_env_name
+    source myenv_$virtual_env_name
+    popd
+}
+
+####################################
+#
+# 1. Modify FIT config files , to  using actual DHCP Host IP instead of 172.31.128.1
+#
+##################################
+setupTestsConfig(){
+    echo "SetupTestsConfig ...replace the 172.31.128.1 IP in test configs with actual DHCP port IP"
+    RACKHD_DHCP_HOST_IP=$(ifconfig | awk '/inet addr/{print substr($2,6)}' |grep 172.31.128)
+    if [ "$RACKHD_DHCP_HOST_IP" == "" ]; then
+         echo "[Error] There should be a NIC with 172.31.128.xxx IP in your OS."
+         exit -2
+    fi
+    pushd ${RACKHD_TEST_DIR}/config
+    sed -i "s/\"username\": \"vagrant\"/\"username\": \"${USER}\"/g" credentials_default.json
+    sed -i "s/\"password\": \"vagrant\"/\"password\": \"${PASSWORD}\"/g" credentials_default.json
+    popd
+    pushd ${RACKHD_TEST_DIR}
+    find ./ -type f -exec sed -i -e "s/172.31.128.1/${RACKHD_DHCP_HOST_IP}/g" {} \;
+    popd
+}
+
+####################################
+#
+# Collect the test report
+#
+##################################
+collectTestReport()
+{
+    mkdir -p ${WORKSPACE}/xunit-reports
+    cp ${RACKHD_TEST_DIR}/*.xml ${WORKSPACE}/xunit-reports
+}
+
+
+####################################
+#
+# Start to run FIT tests
+#
+##################################
+runFIT() {
+    set +e
+    netstat -ntlp
+    pushd ${RACKHD_TEST_DIR}
+    echo "########### Run FIT Smoke Test #############"
+    python run_tests.py ${TEST_GROUP} -stack docker_local_run -v 4 -xunit
```

Comment:
because the stack = docker_local_run
Please clear state the comments in the usage  :   this script only supports test RackHD running in docker with specific config.json .